README.md

```diff
@@ -123,6 +141,73 @@ will recursively (including subdirectories) load all the YAML files. For example
         headers:
           Content-Type: text/html
 ```
+
+### Templates
+Templates can be useful to assemble your payloads from parts
+
+```yaml
+- key: dog
+  kind: Template
+  template: >
+    <animal>dog</animal>
+
+- key: cat
+  kind: Template
+  template: >
+    <animal>cat</animal>
+
+- key: get-freds-pets
+  kind: Behavior
+  expect:
+    http:
+      method: GET
+      path: /fred
+  actions:
+    - reply_http:
+        status_code: 200
+        body: >
+          <human>
+            <name>fred</name>
+            <pets>
+              {{template "dog" .}}
+              {{template "cat" .}}
+            </pets>
+          </human>
+```
+
+### Abstract Behaviors
+Abstract Behaviors can be used to parameterize some data
+
+```yaml
+- key: fruit-of-the-day
+  kind: AbstractBehavior
+  values:
```

Comment:
does `day: whatever` need to be included in this example?

Reply:
no - this makes `fruit` essentially an optional parameter, while `day` is required

Reply:
got it